addons/openebs/1.12.0/install.sh

```diff
@@ -256,3 +259,149 @@ function openebs_cstor_pool_count() {
 }
+
+function openebs_upgrade() {
+    # TODO: handle namespace spec changes at the upgrade time
+    if [ "$OPENEBS_CSTOR" = "0" ]; then
+        # upgrades only required for cStor 
+        return 0
+    fi
+
+    # RE: https://github.com/openebs/openebs/tree/master/k8s/upgrades/1.x.0-1.12.x
+    # Upgrading a minor version of 1.x.x requires two jobs to update pools a volumes for cStor configurations.
+    local runningVer=$(kubectl -n $OPENEBS_NAMESPACE get deploy openebs-provisioner -o jsonpath='{.metadata.labels.openebs\.io/version}')
+
+    local semVerRunningList=( ${runningVer//./ } )
+    local semVerInstallList=( ${OPENEBS_VERSION//./ } )
+    
+    if [[ ${semVerInstallList[0]} -gt 1 ]]; then
+        bail "Only upgrades upto OpenEBS 1.12.0 are tested and supported."
+    fi
+
+    logSubstep "Runnig upgrade checks..."
+    if [[ ${semVerInstallList[1]} -gt ${semVerRunningList[1]} || openebs_check_pools ]]; then
+        logSubstep "As part of OpenEBS upgrade pools and volumes need to be upgraded. At least some of the pools and(or) volumes require an update to match OpenEBS controll plane."
+        logFail "It is highly recommended to schedule a downtime for the application using the OpenEBS PV while performing this upgrade. Also, make sure you have taken a backup of the data before starting the below upgrade procedure."
+        printf "Continue? "
+        if ! confirmN " "; then
+            bail "Will not upgrade OpenEBS. Modify your spec and re-run instllation."
+        fi
+
+        openebs_upgrade_pools
+        openebs_upgrade_vols
+    fi
+}
+
+function openebs_upgrade_pools() {
+    # NOTE: slightly different arguments to pass for pre and after 1.9.0.
+    # Since we only support 1.6.0 -> 1.12.0 using prefixs for pre 1.9.0.
+    local manifest="/tmp/openebs_pool_job.yaml"
+    local pools=$(kubectl get spc --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}')
+    
+    # Bulk upgrade only supported for versions >=1.9.0.
+    # Have to create a job per pool.
+    for pool in $pools; do
+        local spcCurrentVer=$(kubectl get spc $pool -o jsonpath='{.versionDetails.status.current}')
+        logSubstep "Upgrading $pool from $spcCurrentVer to $OPENEBS_VERSION"
+        local out_file=/tmp/openebs-pool-$pool.yaml
+        cat <<UPGRADE_POOLS >$out_file
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: cstor-spc-$RANDOM
+  namespace: $OPENEBS_NAMESPACE
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-maya-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+        - "--from-version=$spcCurrentVer"
+        - "--to-version=$OPENEBS_VERSION"
+        - "--spc-name=$pool"
+        - "--v=4"
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+        image: quay.io/openebs/m-upgrade:$OPENEBS_VERSION
+        imagePullPolicy: Always
```

Comment:
This will break airgap

Reply:
Good catch, thank you. Will work on adding the image to the `Manifest`

Reply:
why do we need `imagepullpolicy: Always` in any situation, though?

Reply:
We can't have anything with imagePullPolicy of Always because that will try to pull in airgap.

Reply:
And I think you missed the main objection - `imagePullPolicy: Always` will ask k8s to pull the image fresh from the internet. This does not work on airgap

Reply:
Changed to  `imagePullPolicy: IfNotPresent`